### PR TITLE
[TTAHUB-3346] Create frontend for QA dashboard main landing page

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- Added widgets to the QA dashboard
- Included a fair amount of refactoring to DRY up some of our widget/graph functionality

## How to test
- Confirm that the widgets on the QA dashboard, regional activity reports dashboard, and regional training reports dashboard all function appropriately and look correct

## Issue(s)
* https://jira.acf.gov/browse/TTAHUB-3346

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/Menu.js [hunk 1]
```diff
@@ -152,7 +152,7 @@ function Menu({
         <ul className="usa-list usa-list--unstyled" role="menu">
           {menuItems.map((item) => (
             <li key={item.label} role="menuitem">
-              <Button type="button" onClick={() => { updateShown(false); item.onClick(); }} unstyled className="smart-hub--menu-button smart-hub--button__no-margin">
+              <Button type="button" id={item.id || ''} onClick={() => { updateShown(false); item.onClick(); }} unstyled className="smart-hub--menu-button smart-hub--button__no-margin">
```
Comment (AdamAdHocTeam) [inline frontend/src/components/Menu.js:155]: Does it make sense for uuid if its undefined?

Comment (thewatermethod) [reply frontend/src/components/Menu.js:155]: why

File: frontend/src/widgets/HorizontalTableWidget.scss [hunk 2]
```diff
@@ -13,10 +13,11 @@
     position: sticky;
     z-index: 1;
     max-width: 330px;
-    min-width: 330px;
+    // min-width: 330px;  I can't figure out why we need this and it's messing me up
```
Comment (AdamAdHocTeam) [inline frontend/src/widgets/HorizontalTableWidget.scss:16]: Can always put it back later if it messes something up.

File: frontend/src/widgets/LineGraph.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                      }
+                      return l;
+                    });
+                    setLegends(updatedLegends);
+                  }}
+                  shape={legend.shape}
+                />
+              ))}
+            </LegendControlFieldset>
+            <div data-testid="lines" ref={lines} />
```
Comment (AdamAdHocTeam) [inline frontend/src/widgets/LineGraph.js:240]: div with nothing in it?

**Review by AdamAdHocTeam — APPROVED** [review]
Nice work, also cleaned up a bunch of stuff.

Might want one more pair of eyes as there are a lot of changes.

File: frontend/src/widgets/LineGraph.js (hunk 3 above)
Comment (thewatermethod) [reply frontend/src/widgets/LineGraph.js:240]: It ends up having a graph in it

**Pull request merged**